File: RPG.Core/Entities/Campaigns/Campaign.cs
using RPG.Core.Entities.Characters;

namespace RPG.Core.Entities.Campaigns;

public class Campaign
{
    public int Id { get; set; }
    public string Name { get; set; }
    public User Owner { get; set; }
    public List<Character> Characters { get; set; } = new();
    public List<CampaignAction> CampaignActions { get; set; } = new();
}

File: client/src/types/campaign.ts
import type { CharacterSummary } from './character.ts'
import type { Owner } from './user.ts'
export interface CampaignInfo {
    id: number;
    name: string;
    campaignCode: string;
    characters: Array<CharacterSummary>;
    owner: Owner;
}

export interface Campaign {
    id: number
    name: string
    campaignCode: string
}

export interface CampaignAction {
    id: number;
    narrative: string;
    actionType: string;
    result: string;
    timestamp: string;
}

export interface PlayableCampaignInfo {
    id: number;
    name: string;
    campaignCode: string;
    characters: Array<CharacterSummary>;
    campaignActions: Array<CampaignAction>;
}